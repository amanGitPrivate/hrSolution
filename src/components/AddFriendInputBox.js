import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './AddFriendInputBox.css';

const AddFriendInputBox = props => {
  const { name, onChange, triggerSubmitOnEnter } = props;
  return (
    <div>
      <input
        type="text"
        autoFocus={true}
        name="name"
        className={classnames('form-control', styles.addFriendInputBox)}
        placeholder="Type the name of a friend"
        value={name}
        onChange={onChange}
        onKeyDown={triggerSubmitOnEnter}
      />
    </div>
  );
};

AddFriendInputBox.propTypes = {
  onChange: PropTypes.func.isRequired,
  triggerSubmitOnEnter: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired
};

export default AddFriendInputBox;
