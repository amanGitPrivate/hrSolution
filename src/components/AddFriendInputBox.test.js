import React from 'react';
import AddFriendInputBox from './AddFriendInputBox';
import renderer from 'react-test-renderer';

test('AddFriendInputBox renders correctly', () => {
  const component = renderer.create(
    <AddFriendInputBox
      name={'Aman'}
      triggerSubmitOnEnter={jest.fn()}
      onChange={jest.fn()}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
