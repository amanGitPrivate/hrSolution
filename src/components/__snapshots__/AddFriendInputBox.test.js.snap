exports[`test AddFriendInputBox renders correctly 1`] = `
<div>
  <input
    autoFocus={true}
    className="form-control"
    name="name"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Type the name of a friend"
    type="text"
    value="Aman" />
</div>
`;
